### PR TITLE
#20466 refactoring offline mode

### PR DESCRIPTION
**Pull request by ovidiuvladpop**

### CODE REVIEW

File: ClearentIdtechIOSFramework/ClearentUI/ClearentProcessingModalViewController.swift [hunk 1]
```diff
@@ -133,6 +128,17 @@ extension ClearentProcessingModalViewController: ClearentProcessingModalView {
 
     // MARK: - Private
     
+    private func shouldDisplayOfflineModeLabel() -> Bool {
+        if ClearentWrapper.configuration.enableOfflineMode {
+            let offlineModeConfirmedDuringPayment = [.payment].contains(ClearentWrapper.shared.flowType?.processType) && presenter?.isOfflineModeConfirmed == true
```
Comment (carmenj) [inline ClearentIdtechIOSFramework/ClearentUI/ClearentProcessingModalViewController.swift:133]: can this be written like this?
```suggestion
            let offlineModeConfirmedDuringPayment = .payment == ClearentWrapper.shared.flowType?.processType && presenter?.isOfflineModeConfirmed
```

File: ClearentIdtechIOSFramework/ClearentWrapper/Wrapper/ClearentWrapper.swift [hunk 2]
```diff
@@ -400,7 +400,7 @@ public final class ClearentWrapper : NSObject {
 extension ClearentWrapper: Clearent_Public_IDTech_VP3300_Delegate {
     
     public func successTransactionToken(_ clearentTransactionToken: ClearentTransactionToken!) {
-        guard let saleEntity = saleEntity else { return }
+//        guard let saleEntity = saleEntity else { return }
```
Comment (carmenj) [inline ClearentIdtechIOSFramework/ClearentWrapper/Wrapper/ClearentWrapper.swift:403]: is this intended?

File: ClearentIdtechIOSFramework/ClearentWrapper/Wrapper/ClearentWrapper.swift [hunk 3]
```diff
@@ -413,9 +413,10 @@ extension ClearentWrapper: Clearent_Public_IDTech_VP3300_Delegate {
     }
     
     public func successOfflineTransactionToken(_ clearentTransactionTokenRequestData: Data?) {
-        guard let saleEntity = saleEntity, let cardToken = clearentTransactionTokenRequestData else { return }
+//        guard let saleEntity = saleEntity,
```
Comment (carmenj) [inline ClearentIdtechIOSFramework/ClearentWrapper/Wrapper/ClearentWrapper.swift:416]: is this intended?

File: ClearentIdtechIOSFramework/ClearentWrapper/Wrapper/ClearentWrapper.swift [hunk 4]
```diff
@@ -34,128 +36,117 @@ public final class ClearentWrapper : NSObject {
...
+    
     /**
-     * Method that should be called to enableOfflineMode
-     * @param key, encryption key usedf for encypting offline transactions
+     * Method that should be called to enable offline mode.
+     * @param key, encryption key used for encrypting offline transactions
      */
-    public func enableOfflineMode(key: SymmetricKey) {
+    
+    //remove key from this one
```
Comment (carmenj) [inline ClearentIdtechIOSFramework/ClearentWrapper/Wrapper/ClearentWrapper.swift:97]: leftover comment?

File: ClearentIdtechIOSFramework/ClearentWrapper/Wrapper/ClearentWrapper.swift [hunk 5]
```diff
@@ -34,128 +36,117 @@ public final class ClearentWrapper : NSObject {
...
+            readEnhancedMessages()
+        }
     }
-
+    
     /**
-     * Method that should be called to enableOfflineMode
-     * @param key, encryption key usedf for encypting offline transactions
+     * Method that should be called to enable offline mode.
+     * @param key, encryption key used for encrypting offline transactions
```
Comment (carmenj) [inline ClearentIdtechIOSFramework/ClearentWrapper/Wrapper/ClearentWrapper.swift:94]: is the `key` param still needed?

File: ClearentIdtechIOSFramework/ClearentWrapper/OfflineMode/Crypto.swift [hunk 6]
```diff
@@ -0,0 +1,16 @@
...
+//  ClearentIdtechIOSFramework
+//
+//  Created by Ovidiu Pop on 02.11.2022.
+//  Copyright © 2022 Clearent, L.L.C. All rights reserved.
+//
+
+import Foundation
+import CryptoKit
+
+@objc public class Crypto: NSObject {
```
Comment (carmenj) [inline ClearentIdtechIOSFramework/ClearentWrapper/OfflineMode/Crypto.swift:12]: is this class used?

Comment (ovidiuvladpop) [reply ClearentIdtechIOSFramework/ClearentWrapper/OfflineMode/Crypto.swift:12]: As the SymmetricKey class is not available in Objective-C, this one has to be used by all integrators in order to create a hash Data object needed for the offline mode encryption.

File: ClearentIdtechIOSFramework/ClearentWrapper/Wrapper/ClearentWrapper.swift (hunk 5 above)
Comment (ovidiuvladpop) [reply ClearentIdtechIOSFramework/ClearentWrapper/Wrapper/ClearentWrapper.swift:94]: Not needed anymore

File: ClearentIdtechIOSFramework/ClearentWrapper/Wrapper/ClearentWrapper.swift (hunk 2 above)
Comment (ovidiuvladpop) [reply ClearentIdtechIOSFramework/ClearentWrapper/Wrapper/ClearentWrapper.swift:403]: Updated

File: ClearentIdtechIOSFramework/ClearentWrapper/Wrapper/ClearentWrapper.swift (hunk 3 above)
Comment (ovidiuvladpop) [reply ClearentIdtechIOSFramework/ClearentWrapper/Wrapper/ClearentWrapper.swift:416]: Updated

**Review by carmenj — APPROVED** [review]

**Pull request merged**